---
layout: post
title: Bioinformatics Class Projects
date: '2016-12-19'
categories: FISH546
tags: teaching
---


Another quarter is complete for our [Bioinformatics class](https://github.com/sr320/course-fish546-2016/wiki) , and once again we  learned a bit. 

This was the general plan‚Ä¶


| Week | Description | Reading |  Quiz
| ------ | ----------- | ------ | ----------- |
| [zero](http://sr320.github.io/course-fish546-2016/00-Notes/)   | [Biology](https://github.com/sr320/course-fish546-2016/blob/master/00-Notes/BioInfo_Intro.pdf), Course Framework, Getting set-up   |  Preface xiii-xxv; How to Learn Bioinformatics 1-18 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%200%22%20)
| [one](http://sr320.github.io/course-fish546-2016/01-Notes/)   | [Bash](https://github.com/sr320/course-fish546-2016/blob/master/01-Notes/Tutorials/Navigating-commandline.md), version control, Project Set-up |  Setting Up and Managing a Bioinformatics Project 21-35; Remedial Unix Shell 37-54 |  [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%201%22%20)
| [two](http://sr320.github.io/course-fish546-2016/02-Notes/) | [Jupyter](https://wp.sanger.ac.uk/barrettgroup/2016/03/10/how-jupyter-notebooks-will-improve-your-computational-life/), [Annotation](https://github.com/sr320/course-fish546-2016/blob/master/jupyter-nbs/003-Local-Blast.ipynb)  | Retrieving Bioinformatics Data 109-124, Unix Data tools 125-168 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%202%22%20)
| [three](http://sr320.github.io/course-fish546-2016/03-Notes/)   | Projects  |  Working with Sequence Data 339-354 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%203%22%20)
| [four](http://sr320.github.io/course-fish546-2016/04-Notes/)  | RNA-seq  | Git for Scientists 67-83 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%204%22%20)
| [five](http://owl.fish.washington.edu/fish546/course-fish546-2016_ncRNA_CGE.pptx) | [lncRNA miRNA](https://github.com/sr320/course-fish546-2016/tree/master/05-Notes/papers)  | Working with Remote Machines 57-66 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%205%22%20)
| [six](http://sr320.github.io/course-fish546-2016/06-Notes/)   | DNA methylation  | [Gavery Slides](https://github.com/sr320/course-fish546-2016/blob/master/06-Notes/Bioinformatics_PartB_Bioinformatics.pdf) |[Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%206%22%20)
| [seven](http://sr320.github.io/course-fish546-2016/07-Notes/)   | Genome Browser  |  Working with Alignment Data 355-383, Working with Range Data 329-338 |[Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%207%22%20)
| eight | Holiday  |  :turkey: :turkey: üåΩ üç∞ üíª 
| [nine](http://sr320.github.io/course-fish546-2016/09-Notes/)  | SNP | Bioinformatics Shell Scripting, Writing Pipelines 395-423 | [Questions](https://github.com/sr320/course-fish546-2016/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Aquiz%20milestone%3A%22Week%209%22%20)


And everyone had a course repo

- [hputnam](https://github.com/hputnam/Oly_Oyster_DNA_Methylation)
- [yaaminiv](https://github.com/yaaminiv/yaaminiv-fish546-2016)
- [mmiddleton](https://github.com/mmiddleton/mmiddleton-fish546)
- [aspanjer](https://github.com/aspanjer/RNAseq_Coho)
- [jldimond](https://github.com/jldimond/jldimond-fish546-2016)
- [mfisher5](https://github.com/mfisher5/mf-fish546-2016)
- [nclowell](https://github.com/nclowell/FISH546)
- [laurahspencer](https://github.com/laurahspencer/546-Bioinformatics)
- [Ellior2](https://github.com/Ellior2/Fish-546-Bioinformatics)
- [MeganEDuffy](https://github.com/MeganEDuffy/FISH-546)
- [sr320](https://github.com/sr320/student-fish546-2016)



That resulted into a ‚ÄúFinal Product‚Äù by the end. 

Here are the highlights of everyone‚Äôs efforts.

---
---


### [@MeganEDuffy](https://github.com/MeganEDuffy)  was  quite [committed](http://eagle.fish.washington.edu/cnidarian/skitch/Punch_card_¬∑_MeganEDuffy_FISH-546_1E083027.png), examining the metaproteomic profile in the Eastern Tropical Pacific Oxygen Minimum Zone.

Duffy, Megan (2016): Metaproteomic profile of an OMZ, version 1. figshare.
https://dx.doi.org/10.6084/m9.figshare.4428059.v1
Retrieved: 15 19, Dec 19, 2016 (GMT)

<iframe src="https://widgets.figshare.com/articles/4428059/embed?show_title=1" width="568" height="351" frameborder="0"></iframe>


_The three underlying goals_:

- Complete both database searching based and de novo sequencing protein identifications for a series of seven depths (55 m to 2500 m) from the Eastern Tropical North Pacific (ETNP) oxygen minimum zone. Compare the quality of identifications from each method and with depth.

- Use identified proteins and peptides to get a sense of microbial diversity compared with oxygen concentration.

 - Compare the ability of de novo peptide sequencing and database searching to identify E. coli peptide standards

One of many of the findings:

![pic](https://raw.githubusercontent.com/MeganEDuffy/FISH-546/master/analyses/2016-11-18-062016_ETNP_129.png)

---

### [@laurahspencer](https://github.com/laurahspencer)  annotated the draft Geoduck genome. 

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.203660.svg)](https://doi.org/10.5281/zenodo.203660)

laurahspencer. (2016). laurahspencer/546-Bioinformatics: Final Project: Geoduck Genome Annotation of >70k bp Scaffolds [Data set]. Zenodo. http://doi.org/10.5281/zenodo.203660

This release includes final project files associated with the annotation of large scaffolds of the Panopea generosa (geoduck) genome. Release files include detailed workflows documented in Jupyter Notebooks, intermediate files, and availability of all source data. Products include:

* Subset of the genome with only >70k bp scaffolds
* Blast alignment of transcriptome to genome scaffolds
* Merged file with transcriptome annotations indexed on genome scaffolds; includes Uniprot data, bp sequences, and start & end sites for both transcriptome & genome
* Candidate Transposable elements, via RepeatMasker.
* Candidate Methylation sites (CpGs) - via Galaxy's EMBOSS fuzznuc tool.
* Candidate miRNA sites - via miRBase;
* RNASeq reads from male & female gonad tissue aligned to genome scaffolds - via bowtie2 & samtools
* SNP sites called via samtools & bcftools
* IGV session with tracks from all the above data, linked to URLs on GitHub and lab web server.

![igv](http://eagle.fish.washington.edu/cnidarian/skitch/546-Bioinformatics_09_Visualizing_with_IGV__ipynb_at_master_¬∑_laurahspencer_546-Bioinformatics_1E08352A.png)

If you want to play with the data yourself in IGV, all you have to do is 
File > Load from url 
`https://raw.githubusercontent.com/laurahspencer/546-Bioinformatics/bdb305054204c872ffb3c06f6404b17c3b29517c/2016-10_Geo-Ann-Project/results/09-Pg_70k_tracks.xml`

![gif](https://drops.azureedge.net/drops/files/acc_524195/GbL9?rscd=inline%3B%20filename%3DScreen%2520Capture%2520on%25202016-12-19%2520at%252007-40-41.gif&rsct=image%2Fgif&se=2016-12-19T16%3A10%3A52Z&sig=hONJ9UIF3RE0yLCk2ZlQdIguFuHI5gx6zuFYFQqGL7c%3D&sp=r&sr=b&st=2016-12-19T15%3A10%3A52Z&sv=2013-08-15)

---

### [@mmiddleton](https://github.com/mmiddleton) described a Oncorhynchus mykiss sperm methylome.

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.203574.svg)](https://doi.org/10.5281/zenodo.203574)

mmiddleton. (2016). mmiddleton/mmiddleton-fish546: O.mykiss sperm methylome [Data set]. Zenodo. http://doi.org/10.5281/zenodo.203574

DNA methylation was assessed with Bismark on both the CoGe platform and locally. 

![coge](https://github.com/mmiddleton/mmiddleton-fish546/raw/master/images/Bismark_notebook/CoGe_success_complete1.png)

```
Parameters used to extract methylation information:
Bismark Extractor Version: v0.16.3
Bismark result file: single-end (SAM format)
Output specified: comprehensive


Processed 2779453 lines in total
Total number of methylation call strings processed: 2779453

Final Cytosine Methylation Report
=================================
Total number of C's analysed:	87198646

Total methylated C's in CpG context:	11732428
Total methylated C's in CHG context:	103618
Total methylated C's in CHH context:	227242

Total C to T conversions in CpG context:	2401136
Total C to T conversions in CHG context:	19981462
Total C to T conversions in CHH context:	52752760

C methylated in CpG context:	83.0%
C methylated in CHG context:	0.5%
C methylated in CHH context:	0.4%
```
---

### [@aspanger](https://github.com/aspanjer)  looked at Coho salmon differential expression analysis across a land use gradient in freshwater streams.

> He pushed 18 commits to master and 18 commits to all branches. On master, 1,818 files have changed and there have been 1,420,664 additions and 924,335 deletions.


Coho Salmon (Oncorhynchus kisutch) are heavily dependent on small stream habitat for rearing, spending over a year in these habitats before out migrating. These fresh water habitats are susceptible to anthropogenic impact from structural changes to habitat, temperature effects from increased runoff and reduced canopy cover, flow changes, and an increased presence of toxicants both from point and non-point sources. As such, Coho health can serve of an indication of anthropogenic influence in these habitats and are of great interest due to their cultural and commercial importance. 

This project uses sequencing data from coho collected at four streams spanning an urban impact gradient. The objectives are:

* Construct a de novo transcriptome for a non-sequenced species.
* Use the annotated Rainbow Trout and Atlantic Salmon genomes to annotate the resulting transcriptome.
* Conduct differential expression analysis between habitat conditions with the intent of identifying biomarkers indicative of toxicant exposure, immune deficiencies, and temperature stress.
* Use Gene Ontology analysis to explore the biological function of observed gene expression differences and to help explain the transcriptome in the context of other sequencing work that‚Äôs been completed on coho.


There was lots of data that informed 

![pca](http://eagle.fish.washington.edu/cnidarian/skitch/Fig1_png__7_documents__7_total_pages__1E08403E.png)

and at the end of the quarter, there were several circles.

![revigo](http://eagle.fish.washington.edu/cnidarian/skitch/rev_1E0851F8.png) 

more at 
[GitHub - aspanjer/RNAseq_Coho](https://github.com/aspanjer/RNAseq_Coho)

and this is what the repo structure looks like


Analysis: the results of each analytical step to complete the above objectives (e.g. FastQC results, trimmed sequencing files, etc)
 
BlastDB: the unique species specific blast databases used for annotating the transcriptome
 
Data_raw: the 48 raw PE sequencing files for each of the 24 sampled individual fish. (available on request)
 
Jupyter_notebook: contains the notebooks for my class project. Focuses on the de novo assembly of a coho transcriptome and subsequent differential expression analysis and annotation:
 
DataQC and Trinity Prep
Differential Expression
Annotation
Gene Ontology
 
Scripts: Scripts used for analysis that aren‚Äôt download directly to my computer through Anaconda (these will be documented in my notebook).

Screenshots: pictures for analysis documentation

---

### [mfisher](https://github.com/mfisher5) tackled some RAD-seq on Pacific cod around the Korean peninsula.  Specifically she wanted to clarify population structure of Pacific cod spawning aggregates in South Korean waters.

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.203351.svg)](https://doi.org/10.5281/zenodo.203351)

mfisher5. (2016). mfisher5/mf-fish546-PCod: Pacific Cod RADseq Project, FISH546 [Data set]. Zenodo. http://doi.org/10.5281/zenodo.203351

![korea](http://eagle.fish.washington.edu/cnidarian/skitch/mfisher5_mf-fish546-PCod_at_v1_0_1E0847A9.png)

Lots of fun stuff to see including

![pca](http://eagle.fish.washington.edu/cnidarian/skitch/mf-fish546-PCod_Lanes_1_and_2_combined__Analyses___Results_ipynb_at_master_¬∑_mfisher5_mf-fish546-PCod_1E084860.png)

With a [great readme](https://github.com/mfisher5/mf-fish546-PCod/blob/v1.0/README.md) to take you where you want to go


_from said page_    


 Class Objectives: 
 **(1) Become familiar with the `stacks` pipeline using Lane 1 sequence data**
 
 *[Jupyter notebook](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/Lane1data_full%20stacks%20pipeline.ipynb)*
 <br
 
 
   **(2) Optimize parameters for `stacks` pipeline**
   
        (a) Research previous papers / labmate work on parameters for c and ustacks, and come up with a list of parameter values to test. 
 
 *[Evernote](https://www.evernote.com/shard/s650/sh/122752f1-9ab7-4f9b-a9af-0c3331617436/0c84d3e0f73055a6)*
        
        (b) Determine quantitative measures that will be used for comparing different batches of c/ustacks parameters
        
        (c) Run through stacks with the Lane 1 data
 *Jupyter notebook [p1](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/testing%20stacks/Testing%20stacks%20Parameters%20I%20.ipynb)*, *[p2](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/testing%20stacks/Testing%20stacks%20Parameters%20II.ipynb)*
        
        (d) Analyze output to decide which parameters to use going forward 
 *[Evernote](https://www.evernote.com/shard/s650/sh/138af148-ea28-416e-b79d-2550b2829d50/3dd0a2619d17e859)*
 
 <br
   **(3) Get a preliminary analysis of population structure between 7 populations in the Lanes 1 and 2 data.** *[Here](https://github.com/mfisher5/mf-fish546-2016/blob/master/Diagrams/PopGen_Workflow.md) is a helpful workflow diagram of population structure analysis, including links to examples of what `stacks` output looks like.*
 
       (a) Run through the full `stacks` pipeline with Lanes 1 and 2 data
 *[Jupyter notebook](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/Lanes%201%20and%202%20combined%20pipeline.ipynb)*
 
       (b) Obtain measures of population structure (paired and overall Fst), and conduct Discriminant Analysis of Principle Components (DAPC)
 *[Jupyter notebook](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/Lanes%201%20and%202%20combined%2C%20Analyses%20%2B%20Results.ipynb)*
 <br
 
 
   **(4) Compare sequencing output from 300ng and degraded DNA protocols to determine if these protocols will be used on the rest of the samples.**
   
 *[Jupyter notebook](https://github.com/mfisher5/mf-fish546-PCod/blob/master/notebooks/Lanes%201%20and%202%20combined%2C%20Analyses%20%2B%20Results.ipynb)*

---

### [@nclowell](https://github.com/nclowell) set out to measure adaptive difference within and between cohorts of Pacific cod in the Salish Sea using RADseq data.

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.203291.svg)](https://doi.org/10.5281/zenodo.203291)

Natalie Lowell. (2016). FISH 546 Final Project [Data set]. Zenodo. http://doi.org/10.5281/zenodo.203291

![workflow](https://github.com/nclowell/FISH546/raw/master/Cod-Time-Series-Project/Notebooks/images_for_notebooks/Screen%20Shot%202016-12-14%20at%201.02.11%20PM.png?raw=true)

There is some great stuff in here.  i.e. 
![fst](http://eagle.fish.washington.edu/cnidarian/skitch/FISH546_Cod-Time-Series-Project_Genepop_Results_ipynb_at_master_¬∑_nclowell_FISH546_1E084C1D.png)

And be sure to check out the Readme.md with [a nice reflection of the quarter](https://github.com/nclowell/FISH546/tree/master/Cod-Time-Series-Project#end-of-quarter-progress)

[![start](http://eagle.fish.washington.edu/cnidarian/skitch/FISH546_Cod-Time-Series-Project_at_master_¬∑_nclowell_FISH546_1E084D06.png)](https://github.com/nclowell/FISH546/tree/master/Cod-Time-Series-Project#end-of-quarter-progress)

---

### [@Ellior2](https://github.com/Ellior2) annotated the Pacific oyster _Crassostrea gigas_ proteome

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.202894.svg)](https://doi.org/10.5281/zenodo.202894)

Rhonda Elliott, mmiddleton, & laurahspencer. (2016). Ellior2/Fish-546-Bioinformatics: Characterization of Pacific oyster proteome- Final Draft [Data set]. Zenodo. http://doi.org/10.5281/zenodo.202894

Goals (via [Readme.md](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/V3.0/README.md) ) 

```
 My goals for this course:
 
 1) Identify proteins and their functions in C. gigas proteome
 TAB file with protein names, GO terms, e-values, etc. [C. gigas GO terms](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/C_gigas_GOterms.tabular) 
 JPG with visualization from Revigo based on GO terms. [C. gigas Revigo visualization](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/C_gigas_Revigo.JPG), [Revigo CSV](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/C_gigas_REVIGO.csv)
 
 2) Compare an oyster proteome to another bivalve- the geoduck
 CSV file with table of proteins and GO terms that are shared between these two bivalve species [C. gigas and P. genorosa table](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/Cgig_Pgen_table.xlsx)
 TXT files with unique proteins specific to each organism [C. gigas](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/blastoutput4.txt), [P. genorosa](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/blastoutput_pgen.txt)
 JPG that visualizes the data [Venn Diagram](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/gigas_prot/Venn_oyster_geoduck.JPG)
 
 3) Draw conclusions about differential protein expression in oysters reared at 23C and 29C from 2015 MS/MS data.
 TAB files listing unique proteins, their functions, and expression levels (in this case peak area) for each sample. [23C oysters at Day 5](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/taylor/2015oyster1_23CDay5.tabular), [29C oysters at Day 5](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/taylor/2015oyster2_29CDay5.tabular), [23C oysters at Day 8](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/taylor/2015oyster13_23CDay8.tabular), [29C oysters at Day 8](https://github.com/Ellior2/Fish-546-Bioinformatics/blob/master/analyses/taylor/2015oyster14_29CDay8.tabular)
 JPGs that visualize differences in protein expression between these two treatments
one of many‚Ä¶.
```

![rev](http://eagle.fish.washington.edu/cnidarian/skitch/Fish-546-Bioinformatics_README_md_at_V3_0_¬∑_Ellior2_Fish-546-Bioinformatics_1E08531E.png)

---

### [@yaaminiv](https://github.com/yaaminiv)  did some analysis of Ostrea lurida gonad nucleotide sequences in control and ocean acidification conditions. 

Venkataraman, Yaamini (2016): Analysis of Ostrea lurida gonad nucleotide sequences in control and ocean acidification conditions. figshare.
https://dx.doi.org/10.6084/m9.figshare.4339886.v1
Retrieved: 17 42, Dec 19, 2016 (GMT)

<iframe src="https://widgets.figshare.com/articles/4339886/embed?show_title=1" width="568" height="351" frameborder="0"></iframe>

You will find a nice series of [Jupyter notebooks](https://github.com/yaaminiv/yaaminiv-fish546-2016/tree/master/notebooks) culminating in some enrichment analyses.

![enr](http://eagle.fish.washington.edu/cnidarian/skitch/yaaminiv-fish546-2016_2016-12-12-oly-gonad-OA-part7-REVIGO_ipynb_at_master_¬∑_yaaminiv_yaaminiv-fish546-2016_1E0855D9.png)

---


### _albnl_ [@jldimond](https://github.com/jldimond) completed his project involving analysis of Porites spp. ddRAD-seq and EpiRAD-seq data.

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.192371.svg)](https://doi.org/10.5281/zenodo.192371)

Jay Dimond. (2016). jldimond/jldimond-fish546-2016: Porites.v2 [Data set]. Zenodo. http://doi.org/10.5281/zenodo.192371

![coral](http://eagle.fish.washington.edu/cnidarian/skitch/jldimond_jldimond-fish546-2016__FISH546_repository_1E09B73D.png) 

His goal was to evaluate     
 (1) if there is any genetic structuring of these individuals using the ddRAD data and      
(2) if there is any epigenetic structuring of these individuals using the EpiRAD data. 


Check out is [readme](https://github.com/jldimond/jldimond-fish546-2016/blob/master/readme.md) for an overview.  And here are a few pics to keep you wanting more‚Ä¶

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/jldimond-fish546-2016_EpiRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B4B4.png" alt="jldimond-fish546-2016_EpiRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B4B4.png"/>

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/jldimond-fish546-2016_EpiRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B4CD.png" alt="jldimond-fish546-2016_EpiRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B4CD.png"/>

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/jldimond-fish546-2016_ddRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B502.png" alt="jldimond-fish546-2016_ddRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B502.png"/>

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/jldimond-fish546-2016_ddRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B523.png" alt="jldimond-fish546-2016_ddRAD_analysis_md_at_master_¬∑_jldimond_jldimond-fish546-2016_1E09B523.png"/>

