Aggregating all the Annotations

One of the ultimate goals here is to visualize any aspect of the draft genome available.  I started out with a [simple blastn using the transcriptome](https://sr320.github.io/student-fish546-2016/Oly-Gene-Track/). The protein name is not embedded but have that in the ready.

`http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/Olyv6_blastn-Scaff-10k.gff`

```
SELECT * FROM [roberts.sbr@gmail.com].[Olyv6_blastx_sprot_2016_09]blast left join [roberts.sbr@gmail.com].[uniprot-all0916]uni on blast.Column3 = uni.Column1
```
Since that simple track I have focused on displaying 
- Transposable Elements
- RNA-seq Expression 
- 2bRAD data corresponding to Bisulfite Data.

The TE track was [generated by Sean](https://seanb80.github.io/blog/2016/11/04/Installing-RepeatMasker)

`http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/RM-sb-Ostrea_lurida-Scaff-10k.fa.out.gff`

I created track using an outdated database
`http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/RM-Ostrea_lurida-Scaff-10k.fa.out.gff`

As you can see Sean's appear more robust
![te](https://raw.githubusercontent.com/sr320/student-fish546-2016/master/docs/images/igv-te.gif)

---

The RNA-seq tracks were [generated](https://github.com/sr320/student-fish546-2016/blob/master/jupyter/02.6-Oly-Bowtie.ipynb) by aligning with bowtie, and creating a sorted, indexed bam.

ie
```
!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \
-x ../data/Ostrea_lurida-Scaff-10k-bowtie-index \
-1 /Volumes/web/nightingales/O_lurida/filtered_106A_Female_Mix_GATCAG_L004_R1.fastq.gz \
-2 /Volumes/web/nightingales/O_lurida/filtered_106A_Female_Mix_GATCAG_L004_R2.fastq.gz \
-p 6 \
--very-fast \
-S /Volumes/caviar/wd/2016-11-11/bw-106A_Female_Mix_GATCAG_L004.sam
!samtools view -bS /Volumes/caviar/wd/2016-11-11/bw-106A_Female_Mix_GATCAG_L004.sam \
| samtools sort -o /Volumes/caviar/wd/2016-11-11/bw-106A_Female_Mix_GATCAG_L004.bam
!samtools index /Volumes/caviar/wd/2016-11-11/bw-106A_Female_Mix_GATCAG_L004.bam
```
with output such as 

```
39823239 reads; of these:
  39823239 (100.00%) were paired; of these:
    34600853 (86.89%) aligned concordantly 0 times
    4879196 (12.25%) aligned concordantly exactly 1 time
    343190 (0.86%) aligned concordantly >1 times
    ----
    34600853 pairs aligned concordantly 0 times; of these:
      535609 (1.55%) aligned discordantly 1 time
    ----
    34065244 pairs aligned 0 times concordantly or discordantly; of these:
      68130488 mates make up the pairs; of these:
        65479761 (96.11%) aligned 0 times
        2276137 (3.34%) aligned exactly 1 time
        374590 (0.55%) aligned >1 times
17.79% overall alignment rate
[bam_sort_core] merging from 33 files...
```

This bam was then converted into a bedgraph...

```
%%bash
bedtools genomecov \
-ibam /Volumes/caviar/wd/2016-11-11/bw-108A_Female_Mix_GGCTAC_L004.bam \
-bg | awk '$4 > 9' > /Volumes/caviar/wd/2016-11-14/108A_Female.bedgraph
```

and tiling with `toTDF` (note: the following can be done in commandline)

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/Screenshot_11_14_16__8_24_AM_1DDA1CA8.png" alt="Screenshot_11_14_16__8_24_AM_1DDA1CA8.png"/>

add in a bit of color, tweaks...

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/IGV_1DDA1CEC.png" alt="IGV_1DDA1CEC.png"/>

```
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/106A_Female.bedgraph.tdf
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/106A_Male.bedgraph.tdf
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/108A_Female.bedgraph.tdf
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/108A_Male.bedgraph.tdf
```

---
And last but not least, I have been trying to visualize [2bRAD data that corresponds to the DNA methylation data](https://github.com/RobertsLab/project-olympia.oyster-genomic/wiki/MBD-BSseq-December-2015) we have.

This has been a bit messier, but I first `lazily` ran bowtie on all files with hc and ss prefix. 

```
%%bash
for f in HC*
do
/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \
-x /Users/sr320/git-repos/student-fish546-2016/data/Ostrea_lurida-Scaff-10k-bowtie-index \
-U $f \
-p 8 \
--very-sensitive-local  \
-S /Volumes/caviar/wd/2016-11-12b/$(basename "${f}" _R1_001.fastq.gz).sam
samtools view -bS /Volumes/caviar/wd/2016-11-12b/$(basename "${f}" _R1_001.fastq.gz).sam \
| samtools sort -o /Volumes/caviar/wd/2016-11-12b/$(basename "${f}" _R1_001.fastq.gz).bam
samtools index /Volumes/caviar/wd/2016-11-12b/$(basename "${f}" _R1_001.fastq.gz).bam
2> /Volumes/caviar/wd/2016-11-12b/$(basename "${f}" _R1_001.fastq.gz).err
done
```
output snip

```
1853604 reads; of these:
  1853604 (100.00%) were unpaired; of these:
    1205281 (65.02%) aligned 0 times
    329041 (17.75%) aligned exactly 1 time
    319282 (17.22%) aligned >1 times
34.98% overall alignment rate
2685751 reads; of these:
  2685751 (100.00%) were unpaired; of these:
    1654380 (61.60%) aligned 0 times
    501862 (18.69%) aligned exactly 1 time
    529509 (19.72%) aligned >1 times
38.40% overall alignment rate
1817265 reads; of these:
  1817265 (100.00%) were unpaired; of these:
    1147629 (63.15%) aligned 0 times
    333311 (18.34%) aligned exactly 1 time
    336325 (18.51%) aligned >1 times
36.85% overall alignment rate
```

Then did some cut and pastes

```
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/HC3-1_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/HC3-5_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/HC3-7_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/HC3-10_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/HC3-11_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS2-9*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS2-14*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS2-18*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS3-3_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS3-14_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS3-15_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS3-16_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS3-20_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
cp: /Volumes/caviar/wd/2016-11-12b/SS3-20_*.ba*: No such file or directory
D-128-95-149-192:~ sr320$ cp /Volumes/caviar/wd/2016-11-12b/SS5-15_*.ba* /Volumes/web/halfshell/2016-11-oly-annotation/
D-128-95-149-192:~ sr320$ /Applications/bioinfo/IGVTools/igvtools 
```
(note one is MIA).

I went ahead and merged them all

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/03-Mapping-2bRAD_1DDA1E14.png" alt="03-Mapping-2bRAD_1DDA1E14.png"/>

and created a single bedgraph where coverage had to be 100x.

```
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/2bRAD-BSmatch.bedgraph
```
<img src="http://eagle.fish.washington.edu/cnidarian/skitch/IGV_1DDA1E69.png" alt="IGV_1DDA1E69.png"/>

I also have bam to visualize SNPs

```
http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/2bRAD-BSmatch.bam
```

<img src="http://eagle.fish.washington.edu/cnidarian/skitch/IGV_1DDA1F41.png" alt="IGV_1DDA1F41.png"/>

---
## But wait! you want to see all this for yourself?

In IGV you can create a session file, and assuming all the files are on a public web server, they should render for you too!

Just take this url
`http://owl.fish.washington.edu/halfshell/2016-11-oly-annotation/igv-session_2016-11-14.xml`


![igv-launch](https://raw.githubusercontent.com/sr320/student-fish546-2016/master/docs/images/igv-launch.gif)


